field_chemistry/06_long_map_graph.Rmd

```diff
@@ -265,35 +174,276 @@ plot_long_jw_ch <- function(param, year = 2021,  title = FALSE){
+
+ggarrange(chla_2021, chla_2022, no3_2021, no3_2022, ncol = 2, nrow = 2)
+#ggsave("output/chla_no3_JW_CH_facet_mean_sd.jpg", width = 24, height = 16, dpi = 300)
+```
+
+# 2022 Ratio Map
+
+This section will create animations and plots that show not only the site data but also it's relationship with down/upstream sites. This can be built out with continous data in mind and could live on a shiny app for funders/public to track changes in the watershed. 
+
+## Function to get ratios
+
+This section organizes the data by site and calculates ratios between upstream and downstream sites.
+
+```{r}
+
+
+
+get_ratios_2022 <- function(nutrient){
+
+  
+  nr <- filter(reservoir_chemistry, site_code %in% c(all_res_system, mainstem_sites, "BEAV") & Year == 2022)
+
+    dplyr::select(date = Date,
+         site_code,
+         all_of(nutrient))%>%
+  pivot_wider(names_from = site_code, values_from = nutrient)%>%
+    filter(date %nin% genomic_dates)
+  ## Remove rows were all columns other than Date are NA
+ nr_no_na_rows <- nr[rowSums(is.na(nr)) != (ncol(nr)-1), ]%>%
+   thicken(interval = "week", 
+           start_val = as.Date("2022-05-30"))
+# remove the date column and consolidate the rows by date_week
+  nr_concise <- nr_no_na_rows%>%
+    select(-date)%>%
+    pivot_longer(cols = c(-date_week), 
+                 names_to = "site",
+                 values_to = "value")%>%
+    drop_na()%>%
+  pivot_wider(names_from = site, values_from = value)
+  
+
+    
+  nr_final_df <- nr_concise%>%
+    mutate(JOEI_JOER  = (JOER/JOEI), 
+           JOER_CBRI = (CBRI/JOER), 
+           CBRI_CBRR = (CBRR/CBRI), 
+           CBRR_CHD = (CHD/CBRR), 
+           CHD_JWC = (JWC/CHD), 
+           BRNR_BMD = (BMD/BRNR), 
+           BMD_JWC = (JWC/BMD), 
+           PTRR_PTRO = (PTRO/PTRR), 
+           PTRO_PJW = (PJW/PTRO), 
+           LNGR_LNGO = (LNGO/LNGR), 
+           LNGO_PJW = (PJW/LNGO), 
+           JWC_SLEP = (SLEP/JWC),
+           PJW_SLEP = (SLEP/PJW), 
+           SLEP_PBR = (PBR/SLEP), 
+           PBR_PSF = (PSF/PBR), 
+           COMI_COMR = (COMR/COMI), 
+           COMR_COMO = (COMO/COMR), 
+           COMO_BEAV =  (BEAV/COMO) , 
+           HORI_HORR = (HORR/HORI), 
+           HORR_HORO = (HORO/HORR), 
+           HORO_BEAV = (BEAV/HORO), 
+           BEAV_SFM = (SFM/BEAV), 
+           SFM_PSF = (PSF/SFM), 
+           PSF_PNF = (PNF/PSF), 
+           PNF_PBD = (PBD/PNF)
+           )%>%
+    pivot_longer(., cols = -date_week, names_to = "site_code", values_to = nutrient)
+  
+ return(nr_final_df)
+   
+}
+
+```
+
+## Join ratio and spatial
+
+Functions make the point and ratio data spatial to be used for mapping
+
+```{r}
+
+
+
+make_line_ratio_spatial_flowlines <- function(nr_final_df, nutrient){
+  
+  nutrient_ratios_lines <- nr_final_df %>%
+  inner_join(ratio_flowlines, by = "site_code")%>%
+  st_sf()%>%
+  dplyr::rename(ratio = nutrient )
+
 
+return(nutrient_ratios_lines)
+  
+}
 
+make_point_ratio_spatial <- function(nr_final_df){
```

Comment:
thought for data vis of ratios: there's a lot of information all at once on these, and when it moves by date it is hard for (at least my) brains to register what's actually going on. What if instead of a gradient of colors to determine the ratio, we have a more binary color scheme (higher/lower) and then preserve changes in size of flowlines based on numeric gradient.... BTW - reviewing your "Ratio Animating" issue right now, too!

Reply:
I tried changes in size but ran into issues in areas where space is tight (Chambers Outflow, PJW/JWC -> SLEP).